crates/common/src/provider/runtime_transport.rs

```diff
@@ -332,3 +332,51 @@ fn url_to_file_path(url: &Url) -> Result<PathBuf, ()> {
 fn url_to_file_path(url: &Url) -> Result<PathBuf, ()> {
     url.to_file_path()
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use reqwest::header::HeaderMap;
+
+    #[tokio::test]
+    async fn test_user_agent_header() {
+        let shutdown_notify = Arc::new(tokio::sync::Notify::new());
+        let shutdown_notify_clone = Arc::clone(&shutdown_notify);
+        let listener = tokio::net::TcpListener::bind("0.0.0.0:1337").await.unwrap();
```

Comment:
Should retrieve the address and use that in the URLs below https://docs.rs/tokio/latest/tokio/net/struct.TcpListener.html#method.local_addr